llvm/lib/CodeGen/ReachingDefAnalysis.cpp

```diff
@@ -344,15 +342,10 @@ int ReachingDefAnalysis::getReachingDef(MachineInstr *MI, Register Reg) const {
     int Key = FrameIndex - ObjectIndexBegin;
 
     // Check that there was a reaching def.
-    auto Lookup1 = MBBFrameObjsReachingDefs.find(MBBNumber);
-    if (Lookup1 == MBBFrameObjsReachingDefs.end())
-      return LatestDef;
-    auto &InnerMap = Lookup1->second;
-    auto Lookup2 = InnerMap.find(Key);
-    if (Lookup2 == InnerMap.end())
+    auto Lookup = MBBFrameObjsReachingDefs.find({MBBNumber, Key});
```

Comment:
Doesn't this need to use FrameIndex instead of Key?

Reply:
yes, forgot to update. thanks

Reply:
So there's no testing?

Reply:
There is no testing for a non-zero ObjectIndexBegin, so it ended up behaving as if Key == FrameIndex. Do you want me  to add a test for a non-zero ObjectIndexBegin?

Reply:
@topperc test added. It doesn't matter anymore though since we no longer care about ObjectIndexBegin.